cranelift/codegen/src/isa/mod.rs

```diff
@@ -315,6 +315,15 @@ pub trait TargetIsa: fmt::Display + Send + Sync {
     fn to_capstone(&self) -> Result<capstone::Capstone, capstone::Error> {
         Err(capstone::Error::UnsupportedArch)
     }
+
+    /// Returns whether this ISA has a native fused-multiply-and-add instruction
+    /// for floats.
+    ///
+    /// Currently this only returns false on x86 when some native features are
+    /// not detected.
+    fn has_native_fma(&self) -> bool {
+        true
```

Comment:
I think I'd prefer not to have a default implementation of `has_native_fma`, or it should default to `false`. That seems to me like something that a backend needs to opt in to, not opt out of.